kernel/src/fs/device.rs

```diff
@@ -77,13 +101,13 @@ impl Debug for DeviceId {
 
 impl From<DeviceId> for u64 {
     fn from(value: DeviceId) -> Self {
-        value.0
+        value.as_encoded_u64()
     }
 }
 
 impl From<u64> for DeviceId {
```

Comment:
Okay, I think we should remove these trait usages and require that users employ methods such as `as_encoded_u64` and `from_encoded_u64` instead. These methods should be properly documented to explain where the encoding magic comes from, with proper references to the Linux implementation. The meanings of `.from()` and `.into()` are too obscure, and it's difficult to understand the code without first understanding the Linux implementation details.

However, this is more of an existing issue and is not related to the changes in this PR. I won't comment further on the `DeviceId`. **At least as far as I'm concerned, you don't need to address this comment.**